apps/epic-web/src/pages/buy.tsx

```diff
@@ -53,19 +53,16 @@ const BuyPage: React.FC<
   allowPurchase,
   bonuses,
 }) => {
-  const {redeemableCoupon, RedeemDialogForCoupon, validCoupon} = useCoupon(
-    couponFromCode,
-    {
-      id: products[0].productId,
-      image: {
-        url: 'https://res.cloudinary.com/epic-web/image/upload/v1695972887/coupon_2x.png',
-        width: 132,
-        height: 112,
-      },
-      title: products[0].title as string,
-      description: products[0]?.description,
+  const {validCoupon} = useCoupon(couponFromCode, {
```

Comment:
I imagine this could be taken a step further -- a `CouponProvider` that handles both golden ticket logic and making other kinds of non-golden validated coupon details available to downstream consumers like where `couponId` gets passed in to `<Pricing ...>`